Source/AdUnits/BannerBaseAdUnit.swift

```diff
@@ -0,0 +1,31 @@
+/*   Copyright 2018-2019 Prebid.org, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import Foundation
+
+
+public class BannerBaseAdUnit: AdUnit {
```

Comment:
Why does this have to be a separate class? cant we add these in the AdUnit itself? Why create bannerBase as a subclass of adUnit and have banner as subclass of bannerBase.
This is an optional field hence need not be implemented by other adUnits if not required

Reply:
From [OpenRTB v2.5 ](https://www.iab.com/wp-content/uploads/2016/03/OpenRTB-API-Specification-Version-2-5-FINAL.pdf)

3.2.6 Object: Banner
    api
    ...

3.2.7 Object: Video
    api
    maxbitrate 
    maxduration
    mimes
    playbackmethod
    protocols
    placement
    ...

It means that every `AdUnit` contains both general and unique parameters. 

If super `AdUnit` contains a general `parameters` field, client can set unnecessary attribute. For example, it would be possible to do like:
``` Swift
let bannerAdUnit = BannerAdUnit(configId: "1001-1", size: CGSize(width: 300, height: 250))

let parameters = AdUnit.Parameters()
parameters.api = [Signals.Api.VPAID_1, Signals.Api.VPAID_2]
parameters.maxBitrate = 1500
parameters.minBitrate = 300
parameters.maxDuration = 30
parameters.minDuration = 5
parameters.mimes = ["video/x-flv", "video/mp4"]
parameters.playbackMethod = [Signals.PlaybackMethod.AutoPlaySoundOn, Signals.PlaybackMethod.ClickToPlay]
parameters.protocols = [Signals.Protocols.VAST_2_0, Signals.Protocols.VAST_3_0]
parameters.startDelay = Signals.StartDelay.PreRoll

bannerAdUnit.parameters = parameters
```

As a developer I want to have an access for all parameters that are relevant to specific `AdUnit` instance(like `BannerAdUnit`, `VideoAdUnit`...).

Reply:
I think i got that... but `api` is a param that is used for all adunits (banner, video, interstitial) so why create a subclass for banner?